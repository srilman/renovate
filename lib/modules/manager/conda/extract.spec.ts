import { Fixtures } from '../../../../test/fixtures';
import { extractPackageFile } from '.';

const environmentYml = Fixtures?.get('environment.yml');

describe('modules/manager/conda/extract', () => {
  describe('extractPackageFile', () => {
    it('returns null for empty', () => {
      expect(extractPackageFile('not an environment.yml')).toBeNull();
    });

    it('extracts dependencies', () => {
      const res = extractPackageFile(environmentYml);
      expect(res?.deps.filter((dep) => dep.skipReason)).toHaveLength(9);
      expect(res?.deps).toHaveLength(12);
      expect(res).toMatchSnapshot();
    });
  });
});
