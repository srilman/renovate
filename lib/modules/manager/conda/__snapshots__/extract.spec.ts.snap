// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules/manager/conda/extract extractPackageFile extracts dependencies 1`] = `
Object {
  "datasource": "conda",
  "deps": Array [
    Object {
      "currentValue": "7.1",
      "depName": "click",
      "skipReason": "unsupported",
    },
    Object {
      "depName": "dask",
      "skipReason": "unsupported",
    },
    Object {
      "currentValue": "2022.5.0",
      "currentVersion": "2022.5.0",
      "depName": "main/distributed",
      "packageName": "main/distributed",
      "versioning": "semver",
    },
    Object {
      "currentValue": "2022.5.0",
      "currentVersion": "2022.5.0",
      "depName": "r/distributed",
      "packageName": "r/distributed",
      "versioning": "semver",
    },
    Object {
      "depName": "mamba[subdir=linux-64,version='>=1.0']",
      "skipReason": "unsupported",
    },
    Object {
      "currentValue": "1.0",
      "depName": "matplotlib",
      "skipReason": "unsupported",
    },
    Object {
      "currentValue": "2.7.1",
      "depName": "seaborn",
      "skipReason": "unsupported",
    },
    Object {
      "currentValue": "2.7.1",
      "depName": "graphviz",
      "skipReason": "unsupported",
    },
    Object {
      "currentValue": "2.7.1",
      "currentVersion": "2.7.1",
      "depName": "conda-forge/mkl",
      "packageName": "conda-forge/mkl",
      "versioning": "semver",
    },
    Object {
      "currentValue": "1.2.beta3",
      "depName": "libblas",
      "skipReason": "unsupported",
    },
    Object {
      "currentValue": "2022.5.0a1",
      "depName": "cloudpickle",
      "skipReason": "unsupported",
    },
    Object {
      "currentValue": "3.1.7",
      "depName": "django-extensions",
      "skipReason": "unsupported",
    },
  ],
}
`;
